internal/provider/resource_tfe_audit_trail_token.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+// NOTE: This is a legacy resource and should be migrated to the Plugin
+// Framework if substantial modifications are planned. See
+// docs/new-resources.md if planning to use this code as boilerplate for
+// a new resource.
+
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	tfe "github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceTFEAuditTrailToken() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEAuditTrailTokenCreate,
+		Read:   resourceTFEAuditTrailTokenRead,
+		Delete: resourceTFEAuditTrailTokenDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTFEAuditTrailTokenImporter,
+		},
+
+		CustomizeDiff: customizeDiffIfProviderDefaultOrganizationChanged,
+
+		Schema: map[string]*schema.Schema{
+			"organization": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+
+			"force_regenerate": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"token": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"expired_at": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceTFEAuditTrailTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(ConfiguredClient)
+	auditTrailTokenType := tfe.AuditTrailToken
+
+	// Get the organization name.
+	organization, err := config.schemaOrDefaultOrganization(d)
+	if err != nil {
+		return err
+	}
+
+	readOptions := tfe.OrganizationTokenReadOptions{
+		TokenType: &auditTrailTokenType,
+	}
+	log.Printf("[DEBUG] Check if a token already exists for organization: %s", organization)
```

Comment:
since this resource deals with `audit_trail_token` let's tailor all the log lines towards `audit trail token`, this will help the debugging process whenever a user has to read through the logs. 

Currently, the logs in this resource have similar log lines as the `tfe_organization_token` which might be difficult to distinguish when reading through logs.

```suggestion
	log.Printf("[DEBUG] Check if an audit token already exists for organization: %s", organization)
```